frontend/pkg/frontend/middleware_validatesubscription.go

```diff
@@ -51,8 +51,20 @@ func (s *SubscriptionStateMuxValidator) MiddlewareValidateSubscriptionState(w ht
 		return
 	}
 
-	ctx := ContextWithSubscription(r.Context(), *sub.Subscription)
-	r = r.WithContext(ctx)
+	// For subscription-scoped requests, ARM will provide the tenant ID
+	// in a "x-ms-home-tenant-id" header. But in test environments this
+	// header may not be present, in which case we can try to fudge it
+	// from the SubscriptionDocument.
+	if r.Header.Get(arm.HeaderNameHomeTenantID) == "" {
+		if sub.Subscription != nil &&
+			sub.Subscription.Properties != nil &&
+			sub.Subscription.Properties.TenantId != nil {
+			r.Header.Set(
+				arm.HeaderNameHomeTenantID,
+				*sub.Subscription.Properties.TenantId)
+		}
+	}
```

Comment:
No technical concerns/I think this will work. I'm trying to weigh whether or not we want to fail anyway (basically force non-prod environment clients to provide this header)

Pros:
- We don't unexpectedly use this logic in prod, admittedly this isn't a big deal because it should never happen, the header should always exist in prod.

Cons:
- I'm not sure if it's providing any value to force non-prod clients to do something differently than prod clients
- On the off chance that this header is actually missing in prod, this fallback logic helps avoid returning a 500

So I agree with this direction of keeping the code as-is

Reply:
We could mitigate your concern if we had a flag or environment variable or other means to know for certain whether we're deployed to prod (i.e. requests are actually coming from ARM).  At the moment we can only infer this from request headers.